pkg/proc/native/proc_freebsd.go

```diff
@@ -348,10 +373,34 @@ func (dbp *nativeProcess) resume() error {
 			thread.CurrentBreakpoint.Clear()
 		}
 	}
+	if len(dbp.os.trapThreads) > 0 {
+		// On these threads we have already received a SIGTRAP while stepping on a different thread,
+		// do not resume the process, instead set a breakpoint on them.
```

Comment:
This comment reads a bit confusing as we're not actually setting a breakpoint here but acknowledging breakpoints that have been hit. I know `stop2` calls `SetCurrentBreakpoint` but it's confusing in this context.